### PR TITLE
Document encrypted password command

### PR DESCRIPTION
**Pull request by hedengran**
Documentation for the encrypted password command.
Related PR: https://github.com/neo-technology/neo4j/pull/7126

### CODE REVIEW

File: cypher/cypher-docs/src/docs/dev/glossary.asciidoc [hunk 1]
```diff
@@ -222,7 +222,9 @@ This section comprises a glossary of all the keywords -- grouped by category and
 |<<functions-reverse-list, reverse()>>           | List              | Returns a list in which the order of all elements in the original list have been reversed.
 |<<functions-reverse, reverse()>>                | String            | Returns a string in which the order of all characters in the original string have been reversed.
 |<<functions-right, right()>>                    | String            | Returns a string containing the specified number of rightmost characters of the original string.
-|<<functions-round, round()>>                   | Numeric           | Returns the value of a number rounded to the nearest integer.
+|<<functions-round1, round()>>                   | Numeric           | Returns the value of the given number rounded to the nearest integer, with half-way values always rounded up.
```
Comment (mnd999) [inline cypher/cypher-docs/src/docs/dev/glossary.asciidoc]: This doesn't seem related to encrypted passwords. Has something unrelated got mixed up in this PR?

File: cypher/cypher-docs/src/docs/dev/ql/functions/index.asciidoc [hunk 2]
```diff
@@ -127,7 +127,9 @@ These functions all operate on numerical expressions only, and will return an er
 | <<functions-ceil, ceil()>>   | Returns the smallest floating point number that is greater than or equal to a number and equal to a mathematical integer.
 | <<functions-floor, floor()>> | Returns the largest floating point number that is less than or equal to a number and equal to a mathematical integer.
 | <<functions-rand, rand()>>   | Returns a random floating point number in the range from 0 (inclusive) to 1 (exclusive); i.e. `[0,1)`.
-| <<functions-round, round()>> | Returns the value of a number rounded to the nearest integer.
+|<<functions-round1, round()>>                                    | Returns the value of the given number rounded to the nearest integer, with half-way values always rounded up.
```
Comment (mnd999) [inline cypher/cypher-docs/src/docs/dev/ql/functions/index.asciidoc]: Same, should this be part of this PR?

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityUserAndRoleManagementTest.scala [hunk 3]
```diff
@@ -103,7 +107,11 @@ class SecurityUserAndRoleManagementTest extends DocumentingTest with QueryStatis
       section("Modifying users", "administration-security-users-alter") {
         p("Users can be modified using `ALTER USER`.")
         p("include::user-management-syntax-alter-user.asciidoc[]")
-        p("The `password` can either be a string value or a string parameter, and is not allowed to be identical to the old password.")
+        p("The `password` can either be a string value or a string parameter, and is not allowed to be identical to the old password. " +
```
Comment (mnd999) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityUserAndRoleManagementTest.scala]:
```suggestion
        p("The `password` can either be a string value or a string parameter, and must not be identical to the old password. " +
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityPrivilegesTest.scala [hunk 4]
```diff
@@ -43,7 +43,7 @@ class SecurityPrivilegesTest extends DocumentingTest with QueryStatisticsTestSup
       p(
         """In this document we will often use the terms _'allows'_ and _'enables'_ in seemingly identical ways. However, there is a subtle difference.
           |We will use _'enables'_ to refer to the consequences of <<administration-security-reads, read privileges>> where a restriction will not cause an error,
-          |only a reduction in the apparent graph size. We will use _'allows'_ to refer to the consequence of <<administration-security-write, write privileges>>
+          |only a reduction in the apparent graph size. We will use _'allows'_ to refer to the consequence of <<administration-security-writes, write privileges>>
```
Comment (mnd999) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityPrivilegesTest.scala]: Is this related to this PR?

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/FulltextIndexTest.scala [hunk 5]
```diff
@@ -73,7 +73,7 @@ class FulltextIndexTest extends DocumentingTest {
           |While a composite index applies only to entities that match the indexed label and _all_ of the indexed properties, full-text index will index entities that have at least one of the indexed labels or relationship types, and at least one of the indexed properties.
           |"""
       )
-      p("For information on how to configure full-text indexes, refer to <<operations-manual#index-configuration-fulltext-search,  Operations Manual -> Indexes to support full-text search>>.")
+      p("For information on how to configure full-text indexes, refer to <<operations-manual#index-configuration-fulltext,  Operations Manual -> Indexes to support full-text search>>.")
```
Comment (mnd999) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/FulltextIndexTest.scala]: Is this related to to this PR?

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityUserAndRoleManagementTest.scala [hunk 6]
```diff
@@ -103,7 +107,11 @@ class SecurityUserAndRoleManagementTest extends DocumentingTest with QueryStatis
...
+          "The optional `PLAINTEXT` in `SET PLAINTEXT PASSWORD` has the same behaviour as `SET PASSWORD`. " +
+          "The optional `ENCRYPTED` can be used to update a user's password when the plaintext password is unknown but the encrypted password is available. " +
+          "With `ENCRYPTED`, the password string is expected to be on the format `<encryption-version>,<hash>,<salt>`."
```
Comment (mnd999) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityUserAndRoleManagementTest.scala]:
```suggestion
          "The optional `ENCRYPTED` can be used to update a user's password when the plaintext password is unknown but the encrypted password is available (e.g. from a database backup). " +
          "With `ENCRYPTED`, the password string is expected to be on the format `<encryption-version>,<hash>,<salt>`."
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityUserAndRoleManagementTest.scala [hunk 7]
```diff
@@ -55,7 +55,11 @@ class SecurityUserAndRoleManagementTest extends DocumentingTest with QueryStatis
         p("Users can be created using `CREATE USER`.")
         p("include::user-management-syntax-create-user.asciidoc[]")
         p("If the optional `SET PASSWORD CHANGE [NOT] REQUIRED` is omitted then the default is `CHANGE REQUIRED`. " +
-          "The default for `SET STATUS` is `ACTIVE`. The `password` can either be a string value or a string parameter.")
+          "The default for `SET STATUS` is `ACTIVE`. The `password` can either be a string value or a string parameter. " +
+          "The optional `PLAINTEXT` in `SET PLAINTEXT PASSWORD` has the same behaviour as `SET PASSWORD`. " +
+          "The optional `ENCRYPTED` can be used to create a user when the plaintext password is unknown but the encrypted password is available. " +
```
Comment (mnd999) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityUserAndRoleManagementTest.scala]:
```suggestion
          "The optional `ENCRYPTED` can be used to create a user when the plaintext password is unknown but the encrypted password is available (e.g. from a database backup). " +
```

**Review by mnd999 — CHANGES_REQUESTED** [review]
Looks like a few extra changes may have snuck in. Also, you should probably label this "Depends w. other PR" and link to the code PR as it's currently not building due to the change not being merged yet.

**Comment by hedengran** [comment]
> Looks like a few extra changes may have snuck in. 

There is an additional commit in the PR where I fixed some broken links. Felt unnecessary to have two PRs when it was easily separated as two commits.

Edit: The second commit has now been moved to a separate PR.

**Review by mnd999 — APPROVED** [review]

**Pull request merged**